src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/TseClaimantRepReplyService.java

```diff
@@ -0,0 +1,366 @@
+                    applicationDoc, applicationType.getType());
+
+            caseData.getDocumentCollection().add(docItem);
+
+        } catch (Exception e) {
+            throw new DocumentManagementException(String.format(DOCGEN_ERROR, caseData.getEthosCaseReference()), e);
+        }
+    }
+
+    private static String getApplicationDoc(GenericTseApplicationType applicationType) {
+        if (RESPONDENT_TITLE.equals(applicationType.getApplicant())) {
+            return uk.gov.hmcts.ecm.common.helpers.DocumentHelper.respondentApplicationToDocType(
+                    applicationType.getType());
+        } else {
+            String claimantApplicationType = claimantSelectApplicationToType(applicationType.getType());
+            return uk.gov.hmcts.ecm.common.helpers.DocumentHelper.claimantApplicationTypeToDocType(
+                    claimantApplicationType);
+        }
+    }
+
+    /**
+     * Reply to a TSE application as a respondent, including updating app status, saving the reply and sending emails.
+     *
+     * @param userToken authorisation token to get claimant's email address
+     * @param caseDetails case details
+     * @param caseData case data
+     */
+    public void claimantReplyToTse(String userToken, CaseDetails caseDetails, CaseData caseData) {
+        updateApplicationState(caseData);
+
+        boolean isRespondingToTribunal = isRespondingToTribunal(caseData);
+        saveReplyToApplication(caseData, isRespondingToTribunal);
+
+        if (isRespondingToTribunal) {
+            sendRespondingToTribunalEmails(caseDetails, userToken);
+        } else {
+            sendRespondingToApplicationEmails(caseDetails, userToken);
+        }
+
+        resetReplyToApplicationPage(caseData);
+    }
+
+    /**
+     * Change status of application to updated if there was an unanswered request for information from the admin.
+     *
+     * @param caseData in which the case details are extracted from
+     */
+    void updateApplicationState(CaseData caseData) {
+        GenericTseApplicationType selectedApplicationType = getClaimantRepSelectedApplicationType(caseData);
+        if (selectedApplicationType.getApplicant().equals(RESPONDENT_TITLE)) {
+            if (isRespondingToTribunal(caseData)) {
+                selectedApplicationType.setApplicationState(WAITING_FOR_THE_TRIBUNAL);
+            } else {
+                selectedApplicationType.setApplicationState(UPDATED);
+            }
+        } else if (isRespondingToTribunal(caseData)) {
+            selectedApplicationType.setApplicationState(UPDATED);
+        }
+    }
+
+    /**
+     * Saves the data on the reply page onto the application object.
+     *
+     * @param caseData contains all the case data
+     * @param isRespondingToTribunal determines if responding to the Tribunal's request/order
+     */
+    void saveReplyToApplication(CaseData caseData, boolean isRespondingToTribunal) {
```

Comment:
can we add the access level to these methods please

Reply:
updateApplicationState and saveReplyToApplication methods are directly access in unit tests also. that's why this is package-private. but if we are to make it private, will have to write tests from the higher order methods